### PR TITLE
Refactorings and specs

### PR DESCRIPTION
**Pull request by osahyoun**
- [x] Pass all action params to AK - previously, only a select few were getting through
- [x] Extract mailing ID from posted akid and reference association when creating action.
- [x] Pass source param to AK for action
- [x] Provide full stack spec coverage with VCR for `POST#action` , `POST#donationpush`, `POST#petitionpage`
- [x] pass appropriate payment method to AK. Default is 'cc'. Need to also handle PayPal.

### CODE REVIEW

**Comment by osahyoun** [comment]
no longer wip

File: Guardfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# A sample Guardfile
```
Comment (EricBoersma) [inline Guardfile]: What's the goal with Guardfile here?

File: app/controllers/message_handler_controller.rb [hunk 2]
```diff
@@ -7,22 +8,15 @@ def handle
     # regardless of whether we're doing a `bundle exec` call or we're
     # getting the messages fed to us by SQS via POST to this HTTP endpoint
     # as is the standard for the Worker Tier in Elastic Beanstalk.
-    resp = QueueListener.new.perform(nil, params)
+    response = QueueListener.new.perform(nil, params)
 
     # Tell SQS everything went hunky dory and we can delete the message.
-    send_status_ok
+    render json: response, status: 200
```
Comment (EricBoersma) [inline app/controllers/message_handler_controller.rb]: Are we using the response on the other end?

File: app/workers/queue_listener.rb [hunk 3]
```diff
@@ -1,55 +1,73 @@
...
+    AkPageCreator.create_page(params.merge(page_type: 'petition'))
+    AkPageCreator.create_page(params.merge(page_type: 'donation'))
+  end
+
+  def extract_mailing_id(akid = '')
+    (akid.try(:split, '.') || []).first
+  end
+
+  def client
+    @client ||= ActionKitConnector::Client.new do |c|
```
Comment (EricBoersma) [inline app/workers/queue_listener.rb]: Does it make sense to switch away from using ENV here and switch to Settings like with Champaign, or am I overthinking it?

File: lib/ak/client.rb [hunk 4]
```diff
@@ -1,7 +1,13 @@
 module Ak
   module Client
+    module_function
+
     def client
```
Comment (EricBoersma) [inline lib/ak/client.rb]: Is it possible to DRY this from the same code above?

File: app/controllers/message_handler_controller.rb (hunk 2 above)
Comment (Tuuleh) [inline app/controllers/message_handler_controller.rb]: Does response actually get sent anywhere? Returning 200 should just pop the message off of the queue. Where's response supposed to be sent here and does it get there?

**Comment by EricBoersma** [comment]
There's lots of deleted logic here - did that move into the gem?

**Comment by Tuuleh** [comment]
I'd like to see this deployed to staging and verified that it works before this PR is merged.

File: app/controllers/message_handler_controller.rb (hunk 2 above)
Comment (osahyoun) [inline app/controllers/message_handler_controller.rb]: It's for integration testing - I run expectations against the response. So long as the queue gets back a 200, we're good.

Comment (Tuuleh) [inline app/controllers/message_handler_controller.rb]: :+1:

File: app/workers/queue_listener.rb (hunk 3 above)
Comment (osahyoun) [inline app/workers/queue_listener.rb]: we're using ENV for only these three vars.

File: lib/ak/client.rb (hunk 4 above)
Comment (osahyoun) [inline lib/ak/client.rb]: certainly is.

**Comment by osahyoun** [comment]
@Tuuleh good call - I've not tested this yet on staging. I'll do some solid testing on Monday.

**Comment by osahyoun** [comment]
@EricBoersma 
> There's lots of deleted logic here - did that move into the gem?

Some logic is in the gem, some was moved to champaign, and some code was deleted (mainly stuff that was duplicating validation logic on AK).

**Comment by EricBoersma** [comment]
Cool, I feel generally good about this, but I'd agree with @Tuuleh that doing some testing on Staging would be a good next step. If that's good, I'm good!

File: Guardfile (hunk 1 above)
Comment (osahyoun) [inline Guardfile]: Eric, it's a gem that'll automatically watch and run specs against any file you've made changes to.

Comment (EricBoersma) [inline Guardfile]: Nifty! I'd imagine that to be quite useful.

**Comment by osahyoun** [comment]
@paulaferris @EricBoersma this has been successfully tested on staging for donation and petition actions.

#### A few example AK resource properties on successful actions made on staging

##### Braintree donation - currency as GBP
image

##### PayPal donation - currency as CAD
image

##### Page with tags
image


##### Mailing extracted from AKID for petitionaction
image

##### Source extracted for petitionaction
image

**Pull request merged**